### ISSUE TITLE
New trace capturing with decycling

### ISSUE THREAD

**Issue by spion** [petkaantonov/bluebird#448]
Just wanted to report that there currently seems to be a bug that causes the following... errm, stack trace :D 

```
/bluebird/js/main/captured_trace.js:27
        var stack = nodes[i].stack;
                            ^
TypeError: Cannot read property 'stack' of undefined
    at CapturedTrace.uncycle (/bluebird/js/main/captured_trace.js:27:29)
    at CapturedTrace.attachExtraTrace (/js/main/captured_trace.js:73:10)
    at Promise._attachExtraTrace (/bluebird/js/main/promise.js:679:21)
    at Promise._settlePromiseFromHandler (/bluebird/js/main/promise.js:598:40)
    at Promise._settlePromiseAt (/bluebird/js/main/promise.js:733:18)
    at Promise._settlePromises (/bluebird/js/main/promise.js:850:14)
    at Async._drainQueue (/bluebird/js/main/async.js:78:16)
    at Async._drainQueues (/bluebird/js/main/async.js:88:10)
    at Async.drainQueues (/bluebird/js/main/async.js:13:14)
    at process._tickCallback (node.js:442:13)
```

edit: ~~this might be caused by interaction with BB 1.0, checking~~ nope its not 1.0

**Comment by benjamingr** [comment]
Can you create a reproducing case?

**Comment by spion** [comment]
@benjamingr not yet, will try.

**Comment by spion** [comment]
I can't make a reproducing case, but this is the result of printing `nodes` in the few calls of uncycle before the crash with the stacktrace I pasted above

[ { [Error]
    _parent: { [Error] _parent: [Object], _length: 11 },
    _length: 12 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 10 },
    _length: 11 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 9 },
    _length: 10 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 8 },
    _length: 9 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 7 },
    _length: 8 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 6 },
    _length: 7 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 5 },
    _length: 6 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 4 },
    _length: 5 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 3 },
    _length: 4 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 2 },
    _length: 3 },
  { [Error]
    _parent: { [Error] _parent: undefined, _length: 1 },
    _length: 2 },
  { [Error] _parent: undefined, _length: 1 } ]
[ { [Error]
    _parent: { [Error] _parent: [Object], _length: 5 },
    _length: 6 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 4 },
    _length: 5 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 3 },
    _length: 4 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 2 },
    _length: 3 },
  { [Error]
    _parent: { [Error] _parent: undefined, _length: 1 },
    _length: 2 },
  { [Error] _parent: undefined, _length: 1 } ]
[ { [Error]
    _parent: { [Error] _parent: [Object], _length: 8 },
    _length: 12 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 7 },
    _length: 8 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 6 },
    _length: 7 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 5 },
    _length: 6 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 4 },
    _length: 5 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 3 },
    _length: 4 },
  { [Error]
    _parent: { [Error] _parent: [Object], _length: 2 },
    _length: 3 },
  { [Error]
    _parent: { [Error] _parent: undefined, _length: 1 },
    _length: 2 },
  { [Error] _parent: undefined, _length: 1 },
  ,
  ,
   ]

Seems like some of the nodes in the list are removed without updating the length counts

**Comment by spion** [comment]
I fixed it in #449 but I wasn't able to make a test case :(
